CMakeLists.txt

```diff
@@ -15,22 +15,25 @@ include(GNUInstallDirs)
 set(CMAKE_CXX_STANDARD 11)
 
 # Third Party Dependencies
-find_package(GSL    REQUIRED)
-find_package(Eigen3 3.3 REQUIRED NO_MODULE)
-find_package(Python REQUIRED COMPONENTS Development)
+find_package(GSL           REQUIRED)
+find_package(Eigen3 3.3    REQUIRED NO_MODULE)
+find_package(Python        REQUIRED COMPONENTS Development)
 find_package(nlohmann_json REQUIRED)
 
 # Library setup
 add_library(ale SHARED
             ${CMAKE_CURRENT_SOURCE_DIR}/src/ale.cpp)
+# Alias a scoped target for safer linking in downstream projects
+set(ALE_HEADERS "include/ale.h")
 set_target_properties(ale PROPERTIES
-    VERSION ${PROJECT_VERSION}
-    SOVERSION 1)
-set(ALE_INCLUDE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/include/")
-
+                      VERSION       ${PROJECT_VERSION}
+                      SOVERSION     0
+                      PUBLIC_HEADER ${ALE_HEADERS})
```

Comment:
Right now this doesn't do much, but if we move to a framework on mac then we need to do this.